common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java

```diff
@@ -1,14 +1,32 @@
 package com.railwayteam.railways.mixin;
 
 import com.railwayteam.railways.Railways;
+import com.railwayteam.railways.registry.CRTrackMaterials;
 import com.simibubi.create.content.trains.track.TrackMaterial;
 import net.minecraft.resources.ResourceLocation;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.List;
 
 @Mixin(value = TrackMaterial.class, remap = false)
 public class MixinTrackMaterial {
+    /**
+     * DataFixer for Modded Compat Cherry Tracks.
+     * If it finds a modded cherry track material it'll replace it {@link CRTrackMaterials#CHERRY }
+     * Which is from 1.20 Vanilla.
+     */
+    @Inject(method = "deserialize", at = @At("HEAD"), cancellable = true)
+    private static void snr$updateCherryCompatTracks(String serializedName, CallbackInfoReturnable<TrackMaterial> cir) {
+        List<String> cherryList = List.of("biomesoplenty:cherry", "byg:cherry", "blue_skies:cherry");
```

Comment:
why aren't we handling narrow and wide variants?

Reply:
Track Materials dont have narrow/wide gauges its just a single material all gauges share

Reply:
wrong. gauges are separate materials, mapping from normal to wide is `CRTrackMaterials.WIDE_GAUGE` and ditto for narrow gauge. the material id is `<namespace>:<base_path>_wide` and `<namespace>:<base_path>_narrow`